docs/API/content-type/listing-co.md

```diff
@@ -1234,20 +1234,82 @@ You can check the example for each filter usage below:
 
 ### Filtering by relation
 
-For example, "Show products in category category-1" is possible using [JsonPath](https://github.com/json-path/JsonPath) standard.
-You have to care about encoding url params. For example:
+Assume we have two Content Type Definitions: `Product` and `Category`. 
+There is a relation where a `Product` can belong to multiple `Categories`.
 
-1. Using JsonPath, product category path is `categories[*].dataUrl`
-1. Expected value is `/api/v1/content/categories/category-1`
-1. Raw query: `GET /api/v1/content/products?filters={"categories[*].dataUrl":{"type":"contains","filter":"/api/v1/content/categories/category-1"}}`
-1. Encoded query: `GET /api/v1/content/products?filters=%7B%22categories%5B%2A%5D.dataUrl%22%3A%7B%22type%22%3A%22contains%22%2C%22filter%22%3A%22%2Fapi%2Fv1%2Fcontent%2Fcategories%2Fcategory-1%22%7D%7D`
+!!! Example Product
 
-Only `contains` and `notContains` type filters can be used with filtering by relation.
+    Here is an example of a `Product` object:
 
-!!! Note
-    Even if you list your content objects with hydration enabled, you cannot filter them by the hydrated data.
+    ```json
+    {
+        "id": "2-id",
+        "price": 100,
+        "title": "product-2",
+        "categories": [
+            {
+                "type": "internal",
+                "dataUrl": "/api/v1/content/category/cat-1"
+            }
+        ]
+    }
+    ```
+    { data-search-exclude }
+
+Filtering by the inverse side of the relation is done using the `dataUrl` field, which identifies the relationship. 
+According to [JsonPath](https://github.com/json-path/JsonPath), in the above example, filtering will be based on the `categories[*].dataUrl` field.
+
+The value used for filtering must be the full `dataUrl` of the object, e.g., `/api/v1/content/category/cat-1`.
+
+!!! Example
+    
+    To display products in the `cat-1` category, you can use a filter of type `includes`:
+    ```json
+    {
+        "categories[*].dataUrl": {
+            "type": "includes",
+            "filter": "/api/v1/content/category/cat-1"
+    }
+    ```
+    { data-search-exclude }
+
+    To display products in either the `cat-1` or `cat-2` categories, you can use a filter of type `overlaps`, allowing you to send a list of categories:
+    ```json
+    {
+        "categories[*].dataUrl": {
+            "type": "overlaps",
+            "filter": [
+                "/api/v1/content/category/cat-1",
+                "/api/v1/content/category/cat-2",
+        ]
+    }
+    ```
+    { data-search-exclude }
+
+Only `contains`, `notContains`, `includes` and `overlaps` type filters can be used with filtering by relation.
```

Comment:
We could improve the description of filtering by relation using different type filters by adding variants per filter in `!!! Example section` like it's done for the filters description in previous part of this documentation page. What do you think about the example section presented below?

```suggestion
Filtering by relation can be done using the following filters: `includes`,`overlaps`, `contains` and `notContains`.
See examples below:

!!! Example

    === "includes"

        To display products in the `cat-1` category, you can use a filter of type `includes`:

        ```json
        {
            "categories[*].dataUrl": {
                "type": "includes",
                "filter": "/api/v1/content/category/cat-1"
            }
        }
        ```

        Will return:

        ```json
        {
            "id": "1-id",
            "title": "product-1",
            "categories": [{
                "type": "internal",
                "dataUrl": "/api/v1/content/category/cat-1"
            }],
            "internal": {...}
        }
        ```
        { data-search-exclude }

        Will not return:

        ```json
        {
            "id": "2-id",
            "title": "product-2",
            "categories": [{
                "type": "internal",
                "dataUrl": "/api/v1/content/category/cat-2"
            }],
            "internal": {...}
        }
        ```
        { data-search-exclude }

        ```json
        {
            "id": "3-id",
            "title": "product-3",
            "categories": [{
                "type": "internal",
                "dataUrl": "/api/v1/content/category/cat-3"
            }],
            "internal": {...}
        }
        ```
        { data-search-exclude }

    === "overlaps"

        To display products in either the `cat-1` or `cat-2` categories, you can use a filter of type `overlaps`, allowing you to send a list of categories:

        ```json
        {
            "categories[*].dataUrl": {
                "type": "overlaps",
                "filter": [
                    "/api/v1/content/category/cat-1",
                    "/api/v1/content/category/cat-2",
                ]
            },
            "internal": {...}
        }
        ```
        { data-search-exclude }

        Will return:

        ```json
        [
            {
                "id":"4-id",
                "title": "product-4",
                "categories":[
                    {
                        "type": "internal",
                        "dataUrl":"/api/v1/content/category/cat-1"
                    },
                    {
                        "type": "internal",
                        "dataUrl":"/api/v1/content/category/cat-2"
                    },
                    {
                        "type": "internal",
                        "dataUrl":"/api/v1/content/category/cat-3"
                    },
                ],
                "internal": {...}
            },
            {
                "id":"5-id",
                "title": "product-5",
                "categories":[
                    {
                        "type": "internal",
                        "dataUrl":"/api/v1/content/category/cat-1"
                    },
                    {
                        "type": "internal",
                        "dataUrl":"/api/v1/content/category/cat-4"
                    },
                ],
                "internal": {...}
            }
        ]
        ```
        { data-search-exclude }

        Will not return:

        ```json
        {
            "id": "3-id",
            "title": "product-3",
            "categories": [{
                "type": "internal",
                "dataUrl": "/api/v1/content/category/cat-3"
            }],
            "internal": {...}
        }
        ```
        { data-search-exclude }

    === "contains"

        To display products that belong to a category containing specific ID part, you can use a filter of type `contains`:

        ```json
        {
            "categories[*].dataUrl": {
                "type": "contains",
                "filter": "/content/category/group-a"
            },
            "internal": {...}
        }
        ```
        { data-search-exclude }

        Will return:

        ```json
        {
            "id": "7-id",
            "title": "product-7",
            "categories": [{
                "type": "internal",
                "dataUrl": "/api/v1/content/category/group-a-cat-1"
            }],
            "internal": {...}
        }
        ```
        { data-search-exclude }

        ```json
        {
            "id": "8-id",
            "title": "product-8",
            "categories": [{
                "type": "internal",
                "dataUrl": "/api/v1/content/category/group-a-cat-2"
            }],
            "internal": {...}
        }
        ```
        { data-search-exclude }

        Will not return:

        ```json
        {
            "id": "1-id",
            "title": "product-1",
            "categories": [{
                "type": "internal",
                "dataUrl": "/api/v1/content/category/cat-1"
            }],
            "internal": {...}
        }
        ```

        ```json
        {
            "id": "9-id",
            "title": "product-9",
            "categories": [{
                "type": "internal",
                "dataUrl": "/api/v1/content/category/group-b-cat-1"
            }],
            "internal": {...}
        }
        ```
        { data-search-exclude }

    === "notContains"

        To display products that belong to a categories that do not contain specific ID part, you can use a filter of type `notContains`:

        ```json
        {
            "categories[*].dataUrl": {
                "type": "notContains",
                "filter": "/content/category/group-a"
            },
            "internal": {...}
        }
        ```
        { data-search-exclude }

        Will return:

        ```json
        {
            "id": "1-id",
            "title": "product-1",
            "categories": [{
                "type": "internal",
                "dataUrl": "/api/v1/content/category/cat-1"
            }],
            "internal": {...}
        }
        ```

        ```json
        {
            "id": "9-id",
            "title": "product-9",
            "categories": [{
                "type": "internal",
                "dataUrl": "/api/v1/content/category/group-b-cat-1"
            }],
            "internal": {...}
        }
        ```
        { data-search-exclude }

        Will not return:

        ```json
        {
            "id": "7-id",
            "title": "product-7",
            "categories": [{
                "type": "internal",
                "dataUrl": "/api/v1/content/category/group-a-cat-1"
            }],
            "internal": {...}
        }
        ```
        { data-search-exclude }

        ```json
        {
            "id": "8-id",
            "title": "product-8",
            "categories": [{
                "type": "internal",
                "dataUrl": "/api/v1/content/category/group-a-cat-2"
            }],
            "internal": {...}
        }
        ```
        { data-search-exclude }
```